crates/services/executor/src/executor.rs

```diff
@@ -549,14 +529,50 @@ where
             Ok(())
         };
 
-        while iter.peek().is_some() {
-            for transaction in iter {
-                execute_transaction(&mut *execution_data, transaction)?;
+        let relayed_tx_iter = forced_transactions.into_iter();
+        for transaction in relayed_tx_iter {
+            const RELAYED_GAS_PRICE: Word = 0;
+            let transaction = MaybeCheckedTransaction::CheckedTransaction(transaction);
+            let tx_id = transaction.id(&self.consensus_params.chain_id());
+            match execute_transaction(
+                &mut *execution_data,
+                transaction,
+                RELAYED_GAS_PRICE,
+            ) {
+                Ok(_) => {}
+                Err(err) => {
+                    let event = ExecutorEvent::ForcedTransactionFailed {
+                        id: tx_id.into(),
+                        block_height,
+                        failure: err.to_string(),
+                    };
+                    execution_data.events.push(event);
+                }
             }
+        }
 
-            remaining_gas_limit = block_gas_limit.saturating_sub(execution_data.used_gas);
+        let remaining_gas_limit = block_gas_limit.saturating_sub(execution_data.used_gas);
 
-            iter = source.next(remaining_gas_limit).into_iter().peekable();
+        // L2 originated transactions should be in the `TxSource`. This will be triggered after
+        // all relayed transactions are processed.
+        let mut regular_tx_iter = source.next(remaining_gas_limit).into_iter().peekable();
+        while regular_tx_iter.peek().is_some() {
+            for transaction in regular_tx_iter {
+                let tx_id = transaction.id(&self.consensus_params.chain_id());
+                match execute_transaction(&mut *execution_data, transaction, gas_price) {
+                    Ok(_) => {}
+                    Err(err) => {
+                        execution_data
+                            .skipped_transactions
+                            .push((tx_id, err.clone()));
+                    }
```

Comment:
We only need to add transaction into skipped transaction in production mode. In other cases we need to fail execution of the whole block since it is invalid

<img width="573" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/10e19a6b-3b57-44c9-b2c7-9f349e740a97">